GetStream/stream-chat-swift
# Allow to pin any message

### 🔗 Issue Links

Resolve https://github.com/GetStream/ios-issues-tracking/issues/782

### 🎯 Goal

Allow to pin any message

### 📝 Summary

- Replace `update` with `partial update` on pinning the message
- Update DemoApp pinning logic

### 🎨 Showcase

<img src="https://github.com/GetStream/stream-chat-swift/assets/20794991/0f4d5b43-8612-4cb1-8703-eb0223aa5b38" width=33% height=33%>

### 🧪 Manual Testing Notes

1. Pin the current user's message
2. Pin the participant's message
3. Login as a participant
=> Verify that messages are pinned with correct labels

1. Unpin the current user's message
2. Unpin the participant's message
3. Login as a participant
=> Verify that messages are unpinned

### ☑️ Contributor Checklist


### 🎁 Meme

![gif](https://media.giphy.com/media/v1.Y2lkPTc5MGI3NjExdG0xZjVlZ2huZDd1NGJ2eHdhenlqcnozYjN6YXJhd2k5Nm93YWxmYSZlcD12MV9pbnRlcm5hbF9naWZfYnlfaWQmY3Q9Zw/26ufdipQqU2lhNA4g/giphy.gif)
